### PR TITLE
DAOS-16712 test: Add mdonssd cond check to list_verbose

### PR DESCRIPTION
**Pull request by tanabarr**
In pool query response (performed implicitly as part of dmg pool list),
expect mem_file_bytes to be zero in PMem mode and equal to scm_size in
MD-on-SSD phase-1 mode (when mem-ratio == 1).

Test-tag: hw,medium,ListVerboseTest
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/pool/list_verbose.py [hunk 1]
```diff
@@ -110,7 +132,7 @@ def create_expected(self, pool, scm_free, nvme_free, scm_imbalance,
                     "imbalance": nvme_imbalance
                 },
             ],
-            "mem_file_bytes": scm_size
+            "mem_file_bytes": 0 if self.is_pmem_mode() else scm_size
```
Comment (phender) [inline src/tests/ftest/pool/list_verbose.py:135]: We have a built-in way to determine if the server is configured with MD on SSD:
```suggestion
            "mem_file_bytes": scm_size if self.server_managers[0].manager.job.using_control_metadata else 0
```

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/pool/list_verbose.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/pool/list_verbose.py:135]: +1

Comment (tanabarr) [reply src/tests/ftest/pool/list_verbose.py:135]: great thanks for pointing that out, I was looking but didn't manage to find it

File: src/tests/ftest/pool/list_verbose.py [hunk 2]
```diff
@@ -110,7 +110,8 @@ def create_expected(self, pool, scm_free, nvme_free, scm_imbalance,
                     "imbalance": nvme_imbalance
                 },
             ],
-            "mem_file_bytes": 0
+            "mem_file_bytes": scm_size if \
+            self.server_manager[0].manager.job.using_control_metadata else 0
```
Comment (phender) [inline src/tests/ftest/pool/list_verbose.py:114]: Shouldn't the line continuation be indented?
```suggestion
            "mem_file_bytes": scm_size if \
                self.server_manager[0].manager.job.using_control_metadata else 0
```

**Review by phender — COMMENTED** [review]
The PR needs to be refreshed to resolve the code conflict

File: src/tests/ftest/pool/list_verbose.py (hunk 2 above)
Comment (daltonbohning) [reply src/tests/ftest/pool/list_verbose.py:114]: Normally the GHA lint check would complain about this but looks like most GHA are deleted on this branch

Comment (tanabarr) [reply src/tests/ftest/pool/list_verbose.py:114]: I tried that but the linter was giving me an error about dangling indent

File: src/tests/ftest/control/dmg_pool_query_test.py [hunk 3]
```diff
@@ -98,7 +98,9 @@ def test_pool_query_basic(self):
                     "size": self.params.get("total", path="/run/exp_vals/nvme/*")
                 }
             ],
-            "mem_file_bytes": 0
+            "mem_file_bytes": self.params.get("total",
+                                              path="/run/exp_vals/scm/*") if
+            self.server_manager[0].manager.job.using_control_metadata else 0
```
Comment (phender) [inline src/tests/ftest/control/dmg_pool_query_test.py:103]: Typo:
```suggestion
            self.server_managers[0].manager.job.using_control_metadata else 0
```

File: src/tests/ftest/pool/list_verbose.py [hunk 4]
```diff
@@ -110,7 +110,8 @@ def create_expected(self, pool, scm_free, nvme_free, scm_imbalance,
                     "imbalance": nvme_imbalance
                 },
             ],
-            "mem_file_bytes": scm_size
+            "mem_file_bytes": scm_size if \
+            self.server_manager[0].manager.job.using_control_metadata else 0
```
Comment (phender) [inline src/tests/ftest/pool/list_verbose.py:114]:
```suggestion
            self.server_managers[0].manager.job.using_control_metadata else 0
```

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/pool/list_verbose.py (hunk 2 above)
Comment (tanabarr) [reply src/tests/ftest/pool/list_verbose.py:114]: whoops, just copied and pasted with the typo, will correct

**Comment by tanabarr** [comment]
fixed typo and satisfied pep-8 indentation rules

**Review by phender — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**